### PR TITLE
add growStack hack to reduce morestack in ParseSQL

### PR DESCRIPTION
**Pull request by zhexuany**
According to flame graph, we have `morestack` present when we call `ParseSQL`.  By adding a 16kb at the beginning, we can hack go compiler allocate 32kb stack frame size for a goroutine.

### CODE REVIEW

File: util/hack/hack.go [hunk 1]
```diff
@@ -41,3 +41,14 @@ func Slice(s string) (b []byte) {
 	pbytes.Cap = pstring.Len
 	return
 }
+
+//
+//go:noinline
+func GrowStack() {
```
Comment (ngaut) [inline util/hack/hack.go]: Increasing stack size has some side effects.

Comment (zhexuany) [reply util/hack/hack.go]: I will provide detail benchmark later. This can reduce 3% percent cpu usage.

Comment (ngaut) [reply util/hack/hack.go]: The change will use more memory, we should do some comparison.

Comment (zhexuany) [reply util/hack/hack.go]: name                      old time/op  new time/op  delta
Basic-8                    1.20s ± 0%   1.18s ± 0%   ~     (p=1.000 n=1+1)
TableScan-8                899µs ± 0%   923µs ± 0%   ~     (p=1.000 n=1+1)
ExplainTableScan-8        66.6µs ± 0%  66.6µs ± 0%   ~     (p=1.000 n=1+1)
TableLookup-8              160µs ± 0%   186µs ± 0%   ~     (p=1.000 n=1+1)
ExplainTableLookup-8      89.9µs ± 0%  90.1µs ± 0%   ~     (p=1.000 n=1+1)
StringIndexScan-8          834µs ± 0%   992µs ± 0%   ~     (p=1.000 n=1+1)
ExplainStringIndexScan-8  82.2µs ± 0%  94.2µs ± 0%   ~     (p=1.000 n=1+1)
StringIndexLookup-8        184µs ± 0%   192µs ± 0%   ~     (p=1.000 n=1+1)
IntegerIndexScan-8         984µs ± 0%   980µs ± 0%   ~     (p=1.000 n=1+1)
IntegerIndexLookup-8       195µs ± 0%   193µs ± 0%   ~     (p=1.000 n=1+1)
DecimalIndexScan-8         984µs ± 0%  1020µs ± 0%   ~     (p=1.000 n=1+1)
DecimalIndexLookup-8       196µs ± 0%   213µs ± 0%   ~     (p=1.000 n=1+1)
InsertWithIndex-8          229µs ± 0%   235µs ± 0%   ~     (p=1.000 n=1+1)
InsertNoIndex-8            184µs ± 0%   203µs ± 0%   ~     (p=1.000 n=1+1)
Sort-8                    77.3ms ± 0%  88.3ms ± 0%   ~     (p=1.000 n=1+1)
Join-8                    1.23ms ± 0%  1.31ms ± 0%   ~     (p=1.000 n=1+1)
JoinLimit-8               1.32ms ± 0%  1.47ms ± 0%   ~     (p=1.000 n=1+1)

Above is benchmark result, apparently this operation do not improve its performance.

**Pull request closed without merging**